opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -553,6 +557,16 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
             with self._lock:
                 self.attributes[key] = value
 
+    def get_attribute(self, key: str) -> types.AttributeValue:
+        if not key:
+            logger.warning("invalid key (empty or null)")
+
+        if key not in self.attributes:
+            logger.warning("no attributes for key:%s", str(key))
```

Comment:
Beware of these kind of checks:

```python
class Something:
    def __init__(self, attributes):

        self._attributes = attributes

    def get_attribute(self, key):

        if key not in self._attributes.keys():
            print("wrong key")
            return None

        return self._attributes[key]


Something({1: "lower"}).get_attribute(3).upper()
```
Running that will produce this:
```
wrong key
Traceback (most recent call last):
  File "attribute.py", line 15, in <module>
    Something({1: "lower"}).get_attribute(3).upper()
AttributeError: 'NoneType' object has no attribute 'upper'
```

This is pretty confusing for the end user, somehow there is a report of a key being wrong but the error is not a `KeyError` but an `AttributeError`. I'm not sure is if is really better to hide exceptions like this in the API implementation, since they will possibly lead to another error immediately after, making it harder to debug errors like this one.

Reply:
Do you have any suggestions for how to make this a little less confusing for the end user.